youtube_dl/YoutubeDL.py

```diff
@@ -339,6 +339,7 @@ class YoutubeDL(object):
     _download_retcode = None
     _num_downloads = None
     _screen_file = None
+    archive_list = []
```

Comment:
a `set` will have significant better performance to check membership (line 2114)

Reply:
I seem to be unable to edit the commit for some reason, perhaps it's because this PR is marked as closed?